include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp

```diff
@@ -360,11 +364,153 @@ create_geometric_coarsening_sequence(Triangulation<dim, spacedim> const & fine_t
       // actually create triangulation
       new_tria->create_triangulation(construction_data);
 
+      // save mesh
+      coarse_grid_triangulations[min_level - 1] = new_tria;
+      temp_tria.coarsen_global();
+    }
+
+    // find out about the number of MPI processes in shared memory regions,
+    // which will later be used to construct a coarser triangulation only on a
+    // single process per MPI rank
+    unsigned int const group_size = [&]() {
+      int rank;
+      MPI_Comm_rank(mpi_comm, &rank);
+
+      MPI_Comm comm_shared;
+      MPI_Comm_split_type(mpi_comm, MPI_COMM_TYPE_SHARED, rank, MPI_INFO_NULL, &comm_shared);
+
+      int size_shared;
+      MPI_Comm_size(comm_shared, &size_shared);
+      MPI_Comm_free(&comm_shared);
+
+      // determine maximum, since some shared memory communicators
+      // might not be filed completely
```

Comment:
filed -> filled. Meaning is unclear to me?

Reply:
Rephrased.